pandas/core/groupby/grouper.py

```diff
@@ -835,6 +835,9 @@ def get_grouper(
 
     # if the actual grouper should be obj[key]
     def is_in_axis(key) -> bool:
+        if key is None:
+            return False
```

Comment:
With this change the following (which I think is technically valid) now raises:

```
df = DataFrame({None: [1, 1, 2, 2], 'b': [1, 1, 2, 3], 'c': [4, 5, 6, 7]})
print(df.groupby(by=[None]).sum())

# Without this change:
#    b   c
# 1  2   9
# 2  5  13
```

A few other thoughts...

It looks like this method is the only use of `_is_label_like` which explicitly excludes `None` from being "label-like".

Also, in our tests the `items.get_loc(key)` is never successful in the case of SeriesGroupBy (obj.ndim == 1). In fact, I'm not sure why you'd be looking in the index for key the except maybe axis=1. But axis=1 seems useless when working with a Series.

Reply:
Conerted to draft for now. Can not think of a reason either why we would check the index for a Series

Reply:
I think the right thing to do here is to have `if obj.ndim == 1: return False` inside the not-label-like block. The current behavior of finding something in the Series index leads to a scalar grouper that will always raise because it's not one dimensional:

```
class A:
    def __str__(self):
        return 'cA'
a = A()
ser = pd.DataFrame({'a': [1, 1, a], 'b': [3, 4, 5]}).set_index('a')['b']
gb = ser.groupby([a])
```

raises `ValueError: Grouper for 'cA' not 1-dimensional`, the grouper in this case being the numpy int 5.